File: Programmers/lv2/rug.cpp
#include <bits/stdc++.h>
using namespace std;

vector<int> solution(int brown, int yellow)
{
    vector<int> answer;
    int size = brown + yellow;
    for (int i = 3; i < size; i++)
    {
        if (size % i != 0)
            continue;
        if (i < size / i)
            continue;
        if (size / i < 3)
            continue;
        int len = (i + size / i) * 2 - 4;
        if (len == brown)
        {
            answer.push_back(i);
            answer.push_back(size / i);
            break;
        }
    }
    return answer;
}

int main()
{
    vector<int> result;
    result = solution(8, 1);
    for (int n : result)
        cout << n << " ";
    cout << endl;

    return 0;
}

File: Programmers/lv2/rug.py
def solution(brown, yellow):
    answer = []
    if yellow >= 3:
        for i in range(1, int(yellow/2)+1):
            y_h = i
            y_w = int(yellow/y_h)
            
            if yellow%y_h != 0:
                continue
            if 2*y_h + (y_w+2)*2 == brown:
                answer.append(y_w+2)
                answer.append(i+2)
                break
    else:
        if yellow == 1 and brown == 8:
            answer.append(3)
            answer.append(3)
        elif yellow == 2 and brown == 10:
            answer.append(4)
            answer.append(3)
    print(answer)
    return answer

solution(12, 4)